pkg/util/helper/binding.go

```diff
@@ -397,8 +397,10 @@ func GenerateReplicaRequirements(podTemplate *corev1.PodTemplateSpec) *workv1alp
 
 	if nodeClaim != nil || resourceRequest != nil {
 		return &workv1alpha2.ReplicaRequirements{
-			NodeClaim:       nodeClaim,
-			ResourceRequest: resourceRequest,
+			NodeClaim:         nodeClaim,
+			ResourceRequest:   resourceRequest,
+			Namespace:         podTemplate.Namespace,
+			PriorityClassName: podTemplate.Spec.PriorityClassName,
```

Comment:
I remember priority class name can be left empty in pod template and K8s pod admission hook will set the default priority class name for pod. So this name could be empty.

Do we handle this default value in the plugin?

Reply:
In this MVP, we probably not, otherwise I need to listwatch PriorityClass in the karmada controller. I can add a comment that if user don't specify in the pod spec, priorityClassName will be empty.

Reply:
> priorityClassName will be empty

Then the plugin will be skipped for those without priorityClassName, right?
That's kind of a limitation, we can do that for this MVP.

Reply:
It will handle by the `matchScope` function https://github.com/karmada-io/karmada/pull/4566/files#diff-e489afa02e13c3ea92d007b4a8575045b8695f34cd31539769f2e28c3c0cb322R240-R261

Reply:
> Besides that, I think client should explicitly specify the priorityClass because the default global priorityClass might be different among managed clusters

But we can't determine the action of users. This field is filled by users.

You could try to watch PriorityClass in estimator instead. If we have a request of empty PriorityClass field, the estimator tries to get default value of its cluster and run this plugin.

Reply:
@Garrybest I think client should explicitly specify the priorityClass because the default global priorityClass might be different among managed clusters.

Reply:
I guess you mean that the administrator might set different default priority classes for each cluster, such as by [PriorityClass](https://github.com/kubernetes/kubernetes/blob/1ad143177c365212e53793aff204c6f1c9727d45/staging/src/k8s.io/api/scheduling/v1/types.go#L37-L47). 

So, the default priority in Karmada might not be accurate.